wireapp/wire-ios-mono
# fix: crash on sending message WPB-3911

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-3911" title="WPB-3911" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-3911</a>  Crash when sending a message to a user with one verified and one non-verified device
  </td></table>
  <br />
 






fix: crash on sending message if other user has both verified and unverified devices
----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Application was crashing when sending message on 1-on-1 conversation to another user that has some devices set as verified and some as unverified (from the perspective of self user).

### Causes (Optional)

There was a threading violation. Incorrect object was used to obtain nonce value for logging purposes.

### Solutions

Use message object from current thread.

### Testing

#### How to Test

It should crash in following scenario, although it did not
1. Log-in onto two accounts: A & B
2. Connect accounts together
3. On user A verify user's B device
4. Log in onto account B on one more device
5. Send a message from user A

Crash was easily forced in following scenario
1. Log-in onto two accounts: A & B
2. Connect accounts together
3. Log in onto account B on one more device
4. On user A verify both user's B device
5. Send a message from user A
6. On user A unverify one of user's B devices
7. Send a message from user A

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
